src/services/userService.ts

```diff
@@ -65,5 +66,19 @@ class UserService {
       throw new Error('500');
     }
   }
+
+  public async refreshToken(token: string): Promise<{ accessToken: string; refreshToken: string }> {
+    const decoded = <JwtPayload>jwt.verify(token, this.TOKEN_KEY);
+    try {
+      const user = await this.getUserById(decoded.id);
+      if (!user || token !== user.refreshToken) {
+        throw new Error('Invalid token');
+      }
+      return this.generateTokens(user);
+    } catch (err) {
+      console.log(err);
+      throw new Error('500');

```

Comment:
I'd rethrow the `err` error in this case.
`express` allows you to set up the default error handler, which e.g. in production mode replaces any error with the 500 Internal Server Error response, but in dev displays the error in the response.

![image](https://user-images.githubusercontent.com/9324869/183035530-d61eaf48-4671-489f-ac87-72cf6f8179e7.png)
https://expressjs.com/en/guide/error-handling.html